scripts/export-signupless-reportbacks-to-rogue.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Script to export Reportbacks that do not have a corresponding Signup from drupal into Rogue.
+ *
+ * to run:
+ * drush --script-path=../scripts/ php-script export-signupless-reportbacks-to-rogue.php
+ */
+
+// Get all the signupless reportbacks
+$reportbacks = db_query("SELECT rb.rbid, rb.nid, rb.run_nid, rb.quantity, rb.why_participated, rb.rbid, rb.flagged, rb.uid
+  FROM dosomething.dosomething_reportback rb
+  LEFT JOIN dosomething.dosomething_signup ds on rb.uid = ds.uid AND rb.run_nid = ds.run_nid
+  WHERE ds.sid IS NULL
+  AND rb.rbid NOT IN (SELECT rbid from dosomething_rogue_reportbacks)");
+
+$client = dosomething_rogue_client();
+
+if (!$reportbacks) {
+  echo 'Nothing to migrate!';
+}
+
+// For each reportback, send a request for each reportback file to POST /posts and Rogue will automatically create the signup
+foreach ($reportbacks as $reportback) {
+  echo 'On rbid ' . $reportback->rbid . '...' . PHP_EOL;
+  $sent_at = date('Y-m-d H:i:s');
+
+  // Try to get Northstar ID
+  $northstar_id = dosomething_user_get_northstar_id($reportback->uid, 'drupal_id');
+
+  if (!$northstar_id) {
+    echo "\t" . 'No northstar id, that is terrible ' . $reportback->rbid . PHP_EOL;
+
+    continue;
+  }
+
+  // Get reportback files
+  $photos = db_query('SELECT rbf.fid, rbf.remote_addr, rbf.caption, rbf.status, rbf.reviewed, rbf.reviewer, rbf.source, rbf.status, rblog.timestamp
+                          FROM dosomething_reportback_file rbf
+                          INNER JOIN dosomething_reportback_log rblog ON rbf.fid = substring_index(rblog.files, \',\',-1)
+                          WHERE rbf.rbid = '. $reportback->rbid . '
+                          AND rbf.fid NOT IN (SELECT fid from dosomething_rogue_reportbacks)
+                          GROUP BY rbf.fid')->fetchAll();
+
+  // Just send a signup if there are no photos (there are about 25 of these)
+  if (!$photos) {
+    echo "\t" . 'Reportback ' . $reportback->rbid . ' has no files, sending just a signup' . PHP_EOL;
+
+    $data = [
+      'northstar_id' => $northstar_id,
+      'campaign_id' => $reportback->nid,
+      'campaign_run_id' => $reportback->run_nid,
+      'created_at' => $photo_created_at,
```

Comment:
does this exist yet?

Reply:
@chloealee great catch! I'm surprised this never errored out on me! I'll also want to use the timestamp of the reportback here

Reply:
@chloealee oooo I bet it didn't get mad because it was still set as something, but it's an irrelevant timestamp here

Reply:
@chloealee hmm I thought this comment was in the spot above where I was using `$photo_created_at` and shouldn't have been. In this spot it gets created on line 92 😺

Reply:
@katiecrane I think it was! Weird!